src/pages/dashboard/feedstock/[id].js

```diff
@@ -18,12 +19,31 @@ import {
   Tabs,
   Text,
 } from '@chakra-ui/react'
-import { useRouter } from 'next/router'
 import { GoDatabase, GoTerminal } from 'react-icons/go'
 
-const Feedstock = () => {
-  const router = useRouter()
-  const { id } = router.query
+export const getServerSideProps = async (context) => {
+  const id = context.params.id
+  const specs = await jsonFetcher(`https://api.pangeo-forge.org/feedstocks`)
+  const spec = specs.find((entry) => entry.id === parseInt(id)).spec
+  const url = `https://raw.githubusercontent.com/${spec}/main/feedstock/meta.yaml`
+  const meta = await yamlFetcher(url)
+  return {
+    props: {
+      id: id,
+      title: meta.title,
+      description: meta.description,
+    },
+  }
+}
+
+const Feedstock = ({ id, title, description }) => {
```

Comment:
@katamartin, i realized i could use `getServerSideProps` for server side rendering... what do you think 🤔 ? should I go ahead and move the data fetching from the card generator into the `getServerSideProps`? are there any performance penalties i should be aware of?


<img width="428" alt="Screen Shot 2022-10-18 at 12 09 19 PM" src="https://user-images.githubusercontent.com/13301940/196510469-a8d7b0ed-17cd-4324-a663-58c8f1fba236.png">


https://www.opengraph.xyz/url/https%3A%2F%2Fpangeo-forge-qq0g6loix-pangeo-forge.vercel.app%2Fdashboard%2Ffeedstock%2F79

Reply:
Nice, I think this makes a lot of sense here because this is the built-in Next pattern for blocking rendering on some data fetching! Because the card generator is run as an [edge function](https://vercel.com/docs/concepts/functions/edge-functions), I don't think `getServerSideProps` will work there, but what you're doing there now looks like a good equivalent (fetching the same information within the function).

The main performance tradeoff to keep in mind is when/where the data fetching is happening with `getServerSideProps` and whether renders are being meaningfully delayed. In this case, I think we'll still be loading the page when we would've previously rendered the loading skeleton but I think this is a fine tradeoff to make!

Reply:
Great... Hopefully, this will also improve SEO. i'm going to (1) move the data fetching logic into `getServerSideProps` and (2) enable caching https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#caching-with-server-side-rendering-ssr